torch/csrc/distributed/rpc/testing/faulty_process_group_agent.cpp

```diff
@@ -76,13 +81,45 @@ std::shared_ptr<FutureMessage> FaultyProcessGroupAgent::send(
   }
 }
 
+void FaultyProcessGroupAgent::enqueueSend(SendWork work) {
+  float msgDelay = getDelayForMessage(work.message_.type());
+  if (msgDelay != 0) {
+    // Sleep for the specified delay for the message.
+    std::this_thread::sleep_for(
+        std::chrono::milliseconds(static_cast<int>(msgDelay * 1000)));
+  }
+  ProcessGroupAgent::enqueueSend(std::move(work));
+}
+
 bool FaultyProcessGroupAgent::shouldFailMessage(MessageType type) const {
   // Return true if the input message type is in the messageTypesToFail_ list
   return (
       std::find(messageTypesToFail_.begin(), messageTypesToFail_.end(), type) !=
       messageTypesToFail_.end());
 }
 
+float FaultyProcessGroupAgent::getDelayForMessage(MessageType type) const {
+  return messagesToDelay_.find(type) == messagesToDelay_.end()
+      ? 0
+      : messagesToDelay_.find(type)->second;
+}
+
+// Lazily constructed map that returns string to message type mapping
+const std::unordered_map<std::string, MessageType> FaultyProcessGroupAgent::
```

Comment:
We only use this function to retrieve the MessageTypes, so can we refactor this to just retrieve the MessageType instead of a map from strings to types?